common/djangoapps/enrollment/tests/test_views.py

```diff
@@ -276,6 +283,35 @@ def test_get_enrollment_with_invalid_key(self):
         self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
         self.assertIn("No course ", resp.content)
 
+    def test_enrollment_throttle_for_user(self):
+        """Make sure a user requests do not exceed the maximum number of requests"""
+        rate_limit_config = RateLimitConfiguration.current()
+        rate_limit_config.enabled = True
+        rate_limit_config.save()
+        CourseModeFactory.create(
+            course_id=self.course.id,
+            mode_slug=CourseMode.HONOR,
+            mode_display_name=CourseMode.HONOR,
+        )
+
+        for attempt in range(0, 50):
+            expected_status = status.HTTP_429_TOO_MANY_REQUESTS if attempt >= 40 else status.HTTP_200_OK
+            self._create_enrollment(expected_status=expected_status)
+
+    def test_enrollment_throttle_for_service(self):
+        """Make sure a service can call the enrollment API as many times as needed. """
+        rate_limit_config = RateLimitConfiguration.current()
```

Comment:
If you assign `self.rate_limit_config = RateLimitConfiguration.current()` in the setup, you could remove this duplicated line.